src/d3d9/d3d9_device.cpp

```diff
@@ -2335,13 +2335,24 @@ namespace dxvk {
           break;
 
         case D3DRS_ZWRITEENABLE:
-          if (likely(!Value != !oldValue))
+          if (likely(m_state.depthStencil != nullptr
+            && m_state.renderStates[D3DRS_ZENABLE])) {
+            // Whether we write the depth has been changed => check for hazards
             UpdateActiveHazardsDS(std::numeric_limits<uint32_t>::max());
```

Comment:
Don't we need to handle the disable case here? That's why I had `if (likely(!Value != !oldValue))`

Reply:
Oh derp.